hydromt/vector.py

```diff
@@ -760,6 +764,7 @@ def from_gdf(
             hdrs = gdf.columns
         else:
             hdrs = [geom_name]
+        da = da.reindex({index_dim: gdf.index})
```

Comment:
I think it would be nice to support different flavours of merging the data similar to the `how` argument in [pandas.merge](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.merge.html)

However I'm not sure it makes sense to create a GeoDataset with empty geometries in case the gdf index overlaps with only a fraction of the dataset index_dim. For now we could start supporting the `how='gdf_index'` (use  only keys from gdf_index - this is basically wat you implemented now) and `how='inner'` (use intersection of both indices). You can use the [pandas.index.intersection](https://pandas.pydata.org/docs/reference/api/pandas.Index.intersection.html?highlight=intersection) method.

Reply:
I thought of it as well, I came to the conclusion that it's created from the GeoDataFrame and the array of data is there as a supplement so to speak. So I focused on the fact that there should always be an index from the gdf and not necessarily from the array. 
However if we find this to be too simplistic and there are use cases where how = 'array' is needed, then we could indeed support that.
I will implement a how keyword (for 'gdf_index and inner'), that seems like a good idea. how = 'gdf_index' is indeed now the way is it.

Reply:
Done, for now. Except for how = 'data'.